Add smhi wind gust speed and thunder probability

crallian opened the pull request:
Added the extra attributes:
wind_gust_speed and
thunder_probability
that were already implemented in the underlaying library (joysoftware/pypi_smhi).

Also for the existing extra attribute "cloudiness", it is now added if "is not None" instead of just "if self.cloudiness" which would make it False (and therefore not available) if cloudiness = 0, which I think was a bug. Therefore i don't consider it to be a breaking change?
I found this because thunder probability is often 0... and I first wrote the attribute code exactly as the old one.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adds wind gust speed and thunder probability as extra attributes. These were already implemented in the underlaying library.

I am sorry, but I am not sure how to run the required tests according to the lists below.
I just made the changes in notepad++, uploaded to custom_components folder and it seems to work without any error logs.

Variables/attributes are added, but as I think they are self explaining, and the existing ones are not documented either, I have not proposed any changes to home-assistant.io.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

crallian committed:
Added some extra attributes

Added the extra attributes
wind_gust_speed and
thunder_probability
that were already implemented in the underlaying library (joysoftware
/
pypi_smhi).

Also for the existing extra attribute cloudiness, it is added if "is not None" instead of just "if self.cloudiness" which would make it False (and therefore not available) if cloudiness = 0.

crallian committed:
Trying to solve the style issues

Removed white spaces and changed order of list as suggested by the tests.

crallian committed:
New try to solve the style issues

Removed some more white spaces...

MartinHjelmare commented on homeassistant/components/smhi/weather.py:
```diff
@@ -238,5 +258,11 @@ def forecast(self) -> list:
-            return {ATTR_SMHI_CLOUDINESS: self.cloudiness}
+        extra_attributes = {}
+        if self.cloudiness is not None:
+            extra_attributes.update({ATTR_SMHI_CLOUDINESS: self.cloudiness})
```
We don't need to create a dict and update the attribute dict. Just set the appropriate key in the attribute dict.
```suggestion
            extra_attributes[ATTR_SMHI_CLOUDINESS] = self.cloudiness
```

crallian replied:
Thank you, that is probably better.
I made a commit changing the three extra attributes as suggested.

crallian committed:
Changed dictionary handling as suggested

Changed dictionary handling as suggested by MartinHjelmare.

MartinHjelmare commented:
Please add an assertion of the new attributes here:
https://github.com/home-assistant/core/blob/4e4042a869f0ff3966cd36882f9c57a031fb37ac/tests/components/smhi/test_weather.py#L59

crallian committed:
Updated test

Updated test_weather.py to include the new attributes wind_gust_speed and thunder_probability.

MartinHjelmare reviewed (approved):
Thanks!

crallian committed:
Added missing imports

Added the missing imports
ATTR_SMHI_THUNDER_PROBABILITY,
ATTR_SMHI_WIND_GUST_SPEED,

crallian committed:
Renaming self.thunder to self.thunder_probability and correcting test valuesfor

Renamed the new internal attribute
thunder to thunder_probability, same as the exposed attribute for improved consistency.
Corrected test values according to smhi.json.

crallian committed:
Forgot to change to self.thunder_probability in one place.

sorry.

crallian commented:
@MartinHjelmare Thank you for the patience and for showing me where to find the test data. I really learned a lot, and hope it will go much smoother next time!

MartinHjelmare reviewed (approved):
Great!
